api/rest_openapi.yaml

```diff
@@ -0,0 +1,335 @@
+                            name:
+                              type: string
+                              description: description of the operating system
+                      required:
+                        - id
+                        - status
+                        - timestamp
+                required:
+                  - pagination
+                  - coreDumps
+              examples:
+                example:
+                  value:
+                    pagination:
+                      value:
+                        totalSize: 127
+                        prev: "/core_dumps?minTimestamp={minTimestamp}&maxTimestamp={maxTimestamp}&application={application}&offset={start}&limit={limit}"
+                        self: "/core_dumps?minTimestamp={minTimestamp}&maxTimestamp={maxTimestamp}&application={application}&offset={start}&limit={limit}"
+                        next: "/core_dumps?minTimestamp={minTimestamp}&maxTimestamp={maxTimestamp}&application={application}&offset={start}&limit={limit}"
+                    coreDumps:
+                      - id: "507f1f77bcf86cd799439011"
+                        status: 1
+                        timestamp: 1663355412
+                        appInfo:
+                          value:
+                            name: "auth-service"
+                            version: "1.5.45"
+                        osInfo:
+                          value:
+                            name: "windows 10"
+        400:
+          description: Bad request
+        500:
+          description: Server error
+    delete:
+      summary: Delete all core dumps
+      description: Method removes all core dumps
+      tags:
+        - mutable
+      responses:
+        200:
+          description: Ok
+        500:
+          description: Server error
+
+  /core_dump/{id}:
+    get:
+      summary: Get core dump by id
+      description: Method returns all the information about core dump by id
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: Core dump id
+          schema:
+            type : string
+      tags:
+        - immutable
+      responses:
+        200:
+          description: Ok
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  coreDump:
+                    type: object
+                    properties:
+                      id:
+                        type: string
+                        description: core dump id
+                      data:
+                        type: string
+                        description: core dump data
+                      status:
+                        type: number
+                        description: core dump status
+                        enum:
+                          - 0
+                          - 1
+                          - 2
+                          - 3
```

Comment:
it seems to me that it can be bad approach to make enums in rest api, i think it's better to use string representations of status.